munch-match/src/App.test.js

```diff
@@ -1,49 +1,38 @@
-import { render, fireEvent} from '@testing-library/react';
+import { render, fireEvent } from '@testing-library/react';
 import React from 'react';
 import FoodOptionButtons from './components/foodOptionButtons';
 import Rating from "./components/rating"
 import Results from "./components/resultsLayout"
 import ResultCard from "./components/resultCard"
 import { getResults } from './services/MapsService.js';
+import foodChoices from './data/categories';
 
 // Testing if food buttons work as expected
 describe('FoodOptionButtons Component', () => {
-
-  it('calls onButtonClick with correct type for each button', () => {
-    const onButtonClickMock = jest.fn();
-    const { getByAltText } = render(<FoodOptionButtons onButtonClick={onButtonClickMock} />);
-    
-    const cafeButton = getByAltText('cafe');
-    fireEvent.click(cafeButton);
-    expect(onButtonClickMock).toHaveBeenCalledWith('cafe');
-
-    const burgerButton = getByAltText('burger');
-    fireEvent.click(burgerButton);
-    expect(onButtonClickMock).toHaveBeenCalledWith('burger');
-
-    const sushiButton = getByAltText('sushi');
-    fireEvent.click(sushiButton);
-    expect(onButtonClickMock).toHaveBeenCalledWith('sushi');
-
-    const fruitButton = getByAltText('fruit');
-    fireEvent.click(fruitButton);
-    expect(onButtonClickMock).toHaveBeenCalledWith('fruit');
-    
+  const onButtonClickMock = jest.fn();
+
+  foodChoices.forEach((choice, index) => {
+    it(`calls onButtonClick with correct type for ${choice}`, () => {
+      const { getByAltText } = render(<FoodOptionButtons onButtonClick={onButtonClickMock} />);
```

Comment:
We should move this line before the `forEach` loop starts to avoid rendering the `FoodOptionButtons` in each iteration.

Reply:
Initially I tried that however the tests fail except for the first, I'm assuming putting it before the forEach causes it to only render the first button.

Reply:
Ah mb I didn't see that you moved `it` into the `forEach`. Given that we're testing the whole `FoodOptionButtons` component, I think it makes sense to keep the `it` outside (surrounding all the test code). Then, moving the line should work.

Reply:
Good idea, I have changed that now :)